File: apps/motoboy-pwa/src/deliveryMapLinks.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { deliveryMapSearchUrl } from "./deliveryMapLinks";

test("uses delivery coordinates when opening the map", () => {
  const url = new URL(
    deliveryMapSearchUrl({
      address: "Rua Teste, 123",
      coordinates: { lat: -24.003825377135037, lng: -46.27390399967142 },
    })!,
  );

  assert.equal(url.searchParams.get("query"), "-24.003825377135037,-46.27390399967142");
});

test("uses trimmed delivery address when coordinates are unavailable", () => {
  const url = new URL(
    deliveryMapSearchUrl({
      address: "  Rua Teste, 123  ",
      coordinates: null,
    })!,
  );

  assert.equal(url.searchParams.get("query"), "Rua Teste, 123");
});

test("does not build a map url without coordinates or address", () => {
  assert.equal(deliveryMapSearchUrl({ address: "   ", coordinates: null }), null);
});

File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryMapLinksTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import com.drogsantoantonio.farmadelivery.data.models.CoordinatesDto
import com.drogsantoantonio.farmadelivery.data.models.DeliveryDto
import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryMapLinksTest {
  @Test
  fun `uses delivery coordinates when opening the map`() {
    val uri = deliveryMapUriString(
      delivery(
        address = "Rua Teste, 123",
        coordinates = CoordinatesDto(lat = -24.003825377135037, lng = -46.27390399967142),
      ),
    )

    assertEquals("geo:0,0?q=-24.003825377135037%2C-46.27390399967142", uri)
  }

  @Test
  fun `uses trimmed delivery address when coordinates are unavailable`() {
    val uri = deliveryMapUriString(
      delivery(
        address = "  Rua Teste, 123  ",
        coordinates = null,
      ),
    )

    assertEquals("geo:0,0?q=Rua%20Teste%2C%20123", uri)
  }

  @Test
  fun `does not build map uri without coordinates or address`() {
    val uri = deliveryMapUriString(
      delivery(
        address = "   ",
        coordinates = null,
      ),
    )

    assertEquals(null, uri)
  }

  private fun delivery(address: String, coordinates: CoordinatesDto?): DeliveryDto {
    return DeliveryDto(
      id = "delivery-1",
      publicCode = "FD-001",
      store = "Drogaria Santo Antonio",
      customer = "Cliente Teste",
      phone = "11999999999",
      address = address,
      status = "AGUARDANDO_MOTOBOY",
      courier = "",
      createdAt = "2026-05-15T00:00:00.000Z",
      priority = "NORMAL",
      coordinates = coordinates,
    )
  }
}
